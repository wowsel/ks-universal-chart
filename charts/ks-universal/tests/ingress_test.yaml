suite: test ingress
templates:
  - deployment.yaml
tests:
  - it: should create an ingress with specified configuration
    set:
      generic:
        ingressesGeneral:
          domain: "example.org"
          ingressClassName: "nginx"
          annotations:
            nginx.ingress.kubernetes.io/proxy-body-size: "10m"
      deployments:
        test-app:
          containers:
            main:
              image: nginx
              imageTag: latest
              ports:
                http:
                  containerPort: 80
                  servicePort: 80
          autoCreateService: true
          autoCreateIngress: true
          ingress:
            annotations:
              nginx.ingress.kubernetes.io/ssl-redirect: "true"
            hosts:
              - subdomain: "test"
                paths:
                  - path: "/"
                    pathType: "Prefix"
              - host: "custom.example.com"
                paths:
                  - path: "/api"
                    pathType: "Prefix"
    asserts:
      - matchSnapshot: {} 