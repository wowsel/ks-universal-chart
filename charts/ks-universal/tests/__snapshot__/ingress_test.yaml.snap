should create an ingress with specified configuration:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/component: test-app
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        helm.sh/chart: ks-universal-0.2.14
      name: test-app
    spec:
      selector:
        matchLabels:
          app.kubernetes.io/component: test-app
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: test-app
      template:
        metadata:
          labels:
            app.kubernetes.io/component: test-app
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: test-app
            helm.sh/chart: ks-universal-0.2.14
        spec:
          containers:
            - image: nginx:latest
              name: main
              ports:
                - containerPort: 80
                  name: http
                  protocol: TCP
  2: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        nginx.ingress.kubernetes.io/proxy-body-size: 10m
        nginx.ingress.kubernetes.io/ssl-redirect: "true"
      labels:
        app.kubernetes.io/component: test-app
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        helm.sh/chart: ks-universal-0.2.14
      name: test-app
    spec:
      ingressClassName: nginx
      rules:
        - host: test.example.org
          http:
            paths:
              - backend:
                  service:
                    name: test-app
                    port:
                      name: http
                path: /
                pathType: Prefix
        - host: custom.example.com
          http:
            paths:
              - backend:
                  service:
                    name: test-app
                    port:
                      name: http
                path: /api
                pathType: Prefix
